### PR TITLE
ARTEMIS-3501 Added exception handling on #handleAddMessage to not sto…

### PR DESCRIPTION
**Pull request by AntonRoskvist**
…p broker from starting with currupted messages

### CODE REVIEW

**Comment by AntonRoskvist** [comment]
I've been unable to write a proper test for this as I cannot simulate a journal broken in the "correct" way. This did however solve the issue I was seeing with an actual broken journal. It will discard and log the corrupted message and proceed with loading the rest of the journal

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java [hunk 1]
```diff
@@ -221,23 +221,27 @@ public void handleAddMessage(Map<Long, Map<Long, AddMessageRecord>> queueMap) th
...
 
-            ref.setDeliveryCount(record.getDeliveryCount());
+               ref.setDeliveryCount(record.getDeliveryCount());
 
-            if (scheduledDeliveryTime != 0) {
-               record.getMessage().setScheduledDeliveryTime(0L);
+               if (scheduledDeliveryTime != 0) {
+                  record.getMessage().setScheduledDeliveryTime(0L);
+               }
+            } catch (Throwable e) {
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java:243]: logger.warn(e.getMessage, e); ?


some context about the message that is causing it?

**Comment by clebertsuconic** [comment]
I don't think it will discard.. the message would still be in the journal until the next restart... right?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java (hunk 1 above)
Comment (AntonRoskvist) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java:243]: This is preceded by a massive dump about everything related to the message, so it seemed a bit excessive. An example of that can be seen in the attached log in the ticket, but it includes messageID, why it failed and the entire message bytebuffer (pretty printed)

**Comment by AntonRoskvist** [comment]
At least in my case I end up in #handleNoMessageReferences (from AbstractJournalStorageManager) and the message is removed by:
storageManager.deleteMessage(msg.getMessageID());

And the following log:
AMQ221019: Deleting unreferenced message id=$MESSAGEID from the journal

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java (hunk 1 above)
Comment (AntonRoskvist) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java:243]: (Thrown by org.apache.activemq.artemis.core.message.impl.CoreMessage #onCheckPropertiesError)

Comment (AntonRoskvist) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/PostOfficeJournalLoader.java:243]: Yes that makes sense, sounds good

**Pull request closed without merging**